code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -198,9 +207,27 @@
 /datum/reagent/proc/on_introduced(var/data)
 	return
 
-/datum/reagent/proc/on_removal(var/data)
+/datum/reagent/proc/on_removal(var/amount)
 	return 1
 
+//Called every tick when listed as an addicted chemical
+/datum/reagent/proc/on_withdrawal(var/mob/living/M)
+	if(!holder)
+		return 1
+	if(!M)
+		M = holder.my_atom //Try to find the mob through the holder
+	if(!istype(M)) //Still can't find it, abort
+		return 1
+	if(M.addicted_chems)
+		M.addicted_chems.remove_reagent(src.id, custom_metabolism)
+	tick++
+	real_tick++
+
+//Has to be a reagents datum for on_withdrawal()
+/mob/living/var/datum/reagents/addicted_chems
+//Associative lists for tolerance formatted like (REAGENT_ID = amount)
+/mob/living/var/list/tolerated_chems = list()
```

Comment:
In addition to the other one I already commented on, this should also be initialized in `create_reagents()` rather than init

Reply:
Done